crates/uv/src/lib.rs

```diff
@@ -58,6 +58,11 @@ async fn run(mut cli: Cli) -> Result<ExitStatus> {
         std::env::set_current_dir(directory)?;
     }
 
+    // Set the `UV` variable
+    if let Ok(current_exe) = std::env::current_exe() {
+        std::env::set_var(EnvVars::UV, current_exe);
+    }
+
```

Comment:
We do this eagerly / early so it's available in any child process we might spawn.

I think this is better than special-casing, e.g., `uv run`.

Reply:
Note that in Rust 2024, this function will be [marked `unsafe`](https://doc.rust-lang.org/edition-guide/rust-2024/newly-unsafe-functions.html#stdenvset_var-remove_var).

What does special-casing entail?

Reply:
We'd need to implement passing this through to every subprocess call (or some subset of them).

I think it's safe here, since we're ~single threaded and we're the only one using this variable. We could move it before we start the tokio runtime for additional "safety", but it's more obvious to put it here.

Reply:
I moved it earlier and into an unsafe block

https://github.com/astral-sh/uv/pull/11326/commits/17aab2f1f5302b416e9c869949bed4f1631af285

Reply:
I think marking this as unsafe and putting it there is just "best practice". I have no practical concerns here.